pennylane/ops/qubit/state_preparation.py

```diff
@@ -88,6 +90,29 @@ def compute_decomposition(n, wires):
         """
         return [BasisStatePreparation(n, wires)]
 
+    # pylint:disable=arguments-renamed,invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return True
```

Comment:
Since `has_matrix=True` will hold for every instance of this class, we can actually just make this a class property and also be able to check `qml.BasisState.has_matrix == True`.  Not too important, but just gives us access to more information when its available.